xonsh/parsers/base.py

```diff
@@ -1894,15 +1895,16 @@ def p_testlist_test(self, p):
         """testlist : test"""
         p1 = p[1]
         if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+                                          p1._real_tuple and p1.elts):
             p1.lineno, p1.col_offset = lopen_loc(p1.elts[0])
         p[0] = p1
 
     def p_testlist_single(self, p):
         """testlist : test COMMA"""
         p1 = p[1]
-        if isinstance(p1, ast.Tuple) and (hasattr(p1, '_real_tuple') and
-                                          p1._real_tuple):
+        if isinstance(p1, ast.List) or (isinstance(p1, ast.Tuple) and
+                                        hasattr(p1, '_real_tuple') and
+                                        p1._real_tuple):
```

Comment:
`isinstance(p1, (ast.List, ast.Tuple))` as @mitnk pointed out is the same as
`isinstance(p1, ast.List) or isinstance(p1, ast.Tuple)`.
also `getattr(p1, '_real_tuple', None)` seems more readable, no reason to replicate older code if you know there is a better way that increases code quality.